### ISSUE TITLE
rails_admin throw ActionController::InvalidAuthenticityToken in rails-api 

### ISSUE THREAD

**Issue by buncismamen** [sferik/rails_admin#2919]
rails_admin throw ActionController::InvalidAuthenticityToken when I want to create model

![image](https://user-images.githubusercontent.com/14258779/29773791-283d1a8c-8c29-11e7-8cde-b9eea88c9b28.png)

rails version 5.1.2 with --api mode
rails_admin version 1.2.0

change `config.api_only` from `true` to `false` in `application.rb` fix the problem idk why, anyone care to explain?

**Comment by buncismamen** [comment]
I think its related with this https://github.com/sferik/rails_admin/issues/2830
adding these stack into rails middleware also fix this 
```
# application.rb
config.session_store :cookie_store
config.middleware.use ActionDispatch::Cookies
config.middleware.use ActionDispatch::Session::CookieStore, config.session_options
```
https://github.com/sferik/rails_admin/issues/2830#issuecomment-323622174

**Comment by efrapp** [comment]
Thanks @buncismamen it worked for me!!

**Comment by visheshruparelia** [comment]
Thank you, this fixed it. But I don't get why this worked, can you explain? @buncismamen

**Comment by buncismamen** [comment]
@visheshruparelia because rails-api remove some middleware stacks that rails admin needs.
so you just need to re-added it.

**Comment by mshibuya** [comment]
Fixed by #3006.

**Comment by 1054949000** [comment]
<img width="1740" alt="截屏2022-04-08 下午6 42 28" src="https://user-images.githubusercontent.com/6510093/162420438-90040ba7-4f5c-44d2-9294-cf49929b61b1.png">
<img width="697" alt="截屏2022-04-08 下午6 42 49" src="https://user-images.githubusercontent.com/6510093/162420453-9d70fbdd-4ad7-465f-86ad-56bebc98ee4a.png">


i have the problem: back 422
ActionController::InvalidAuthenticityToken

i have add the code in application.rb

config.session_store :cookie_store
    config.middleware.use ActionDispatch::Cookies
    config.middleware.use ActionDispatch::Session::CookieStore, config.session_options
    config.middleware.use Rack::MethodOverride
    config.api_only = false
